src/EA.Weee.RequestHandlers/Admin/AddPaymentDetailsHandler.cs

```diff
@@ -0,0 +1,57 @@
+﻿namespace EA.Weee.RequestHandlers.Admin
+{
+    using Core.Scheme;
+    using Domain.Scheme;
+    using EA.Prsd.Core.Domain;
+    using EA.Weee.Core.PaymentDetails;
+    using EA.Weee.DataAccess;
+    using EA.Weee.DataAccess.DataAccess;
+    using EA.Weee.Domain.Producer;
+    using EA.Weee.Security;
+    using Prsd.Core.Mapper;
+    using Prsd.Core.Mediator;
+    using Security;
+    using System;
+    using System.Threading.Tasks;
+
+    public class AddPaymentDetailsHandler : IRequestHandler<Requests.Admin.AddPaymentDetails, ManualPaymentResult>
+    {
+        private readonly IWeeeAuthorization authorization;
+        private readonly IGenericDataAccess dataAccess;
+        private readonly IUserContext userContext;
+        private readonly WeeeContext context;
+
+        public AddPaymentDetailsHandler(IWeeeAuthorization authorization, 
+            IGenericDataAccess genericDataAccess,
+            IUserContext userContext,
+            WeeeContext context)
+        {
+            this.authorization = authorization;
+            this.dataAccess = genericDataAccess;
+            this.userContext = userContext;
+            this.context = context;
+        }
+
+        public async Task<ManualPaymentResult> HandleAsync(Requests.Admin.AddPaymentDetails request)
```

Comment:
Should there be tests for this?

Reply:
src/EA.Weee.RequestHandlers.Tests.Unit/Admin/AddPaymentDetailsHandlerTests.cs @will-hope98